code/modules/games/cards/lotto.dm

```diff
@@ -0,0 +1,132 @@
+/obj/item/toy/lotto_ticket
+	name = "scratch-off lotto ticket"
+	desc = "A scratch-off lotto ticket."
+	icon = 'icons/obj/toy.dmi'
+	w_class = W_CLASS_TINY
+	var/revealed = FALSE
+	var/iswinner = FALSE
+	var/ticket_price
+	var/winnings = 0
+
+/obj/item/toy/lotto_ticket/New()
+	..()
+	pixel_y = rand(-8, 8) * PIXEL_MULTIPLIER
+	pixel_x = rand(-9, 9) * PIXEL_MULTIPLIER
+
+/obj/item/toy/lotto_ticket/proc/scratch(var/input_price)
+	var/list/prizelist = list(100000,50000,10000,5000,1000,500,250,100,50,20,10,5,4,3,2,1)
+	var/list/problist = list(0.0001, 0.0002, 0.001, 0.002, 0.01, 0.02, 0.04, 0.2, 1, 2.5, 5, 10, 12.5, 17, 20, 25)
+	var/tuning_value = 1/5 //Used to adjust expected values.
+	var/profit = 0
+	for(var/prize = 1 to problist.len)
+		if(prob(problist[prize]))
+			profit = prizelist[prize]*input_price*tuning_value
+			return profit
+
+//Flash code taken from Blinder
+/obj/item/toy/lotto_ticket/proc/flash(var/turf/T , var/mob/living/M)
+	playsound(src, 'sound/effects/EMPulse.ogg', 100, 1)
+
+	if(M.blinded)
+		return
+
+	M.flash_eyes(visual = 1, affect_silicon = 1)
+
+	if(issilicon(M))
+		M.Knockdown(rand(5, 10))
+		M.visible_message("<span class='warning'>[M]'s sensors are overloaded by the flash of light!</span>","<span class='warning'>Your sensors are overloaded by the flash of light!</span>")
+
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		var/datum/organ/internal/eyes/E = H.internal_organs_by_name["eyes"]
+		if (E && E.damage >= E.min_bruised_damage)
+			to_chat(M, "<span class='warning'>Your eyes start to burn badly!</span>")
+	M.update_icons()
+
+
+/obj/item/toy/lotto_ticket/attackby(obj/item/weapon/S, mob/user)
+	if(!revealed)
```

Comment:
I feel as if an early return would be good here, so go straight to the 
> to_chat(user, "<span class='notice'>The film covering the prizes has already been scratched off.</span>")

Reply:
What do you mean? If !revealed is false (meaning it was already scratched off) then the associated else produces 

>to_chat(user, "<span class='notice'>The film covering the prizes has already been scratched off.</span>")